FAILED RoutersSpec

https://jenkins.akka.io:8498/job/akka-multi-node-nightly/6860/testReport/junit/akka.actor.typed.scaladsl/RoutersSpec/The_router_pool_must_create_n_children_and_route_messages_to/
https://jenkins.akka.io:8498/job/akka-nightly/6361/testReport/
https://jenkins.akka.io:8498/job/akka-multi-node-nightly/6864/
https://jenkins.akka.io:8498/job/akka-nightly-jdk11/167/
https://jenkins.akka.io:8498/job/akka-nightly-2.13/94/

```
java.lang.AssertionError: expected 2 message-2, found 3 message-2
	at akka.actor.testkit.typed.internal.TestProbeImpl.assertFail(TestProbeImpl.scala:375)
	at akka.actor.testkit.typed.internal.TestProbeImpl.expectMessage_internal(TestProbeImpl.scala:161)
	at akka.actor.testkit.typed.internal.TestProbeImpl.expectMessage(TestProbeImpl.scala:143)
	at akka.actor.typed.scaladsl.RoutersSpec.$anonfun$new$6(RoutersSpec.scala:58)
	at akka.actor.typed.scaladsl.RoutersSpec.$anonfun$new$6$adapted(RoutersSpec.scala:55)
	at scala.collection.immutable.Range.foreach(Range.scala:158)
	at akka.actor.typed.scaladsl.RoutersSpec.$anonfun$new$2(RoutersSpec.scala:55)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
	at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
	at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
	at org.scalatest.Transformer.apply(Transformer.scala:22)
	at org.scalatest.Transformer.apply(Transformer.scala:20)
	at org.scalatest.WordSpecLike$$anon$3.apply(WordSpecLike.scala:1075)
	at org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
	at org.scalatest.TestSuite.withFixture$(TestSuite.scala:195)
	at akka.actor.testkit.typed.scaladsl.ScalaTestWithActorTestKit.withFixture(ScalaTestWithActorTestKit.scala:23)
```

--- a/akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/RoutersSpec.scala
+++ b/akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/RoutersSpec.scala
@@ -38,12 +38,13 @@ class RoutersSpec extends ScalaTestWithActorTestKit("""
 
     "create n children and route messages to" in {
       val childCounter = new AtomicInteger(0)
-      val probe = createTestProbe[String]()
+      case class Ack(msg: String, recipient: Int)
+      val probe = createTestProbe[AnyRef]()
       val pool = spawn(Routers.pool[String](4)(Behaviors.setup { _ ⇒
         val id = childCounter.getAndIncrement()
         probe.ref ! s"started $id"
         Behaviors.receiveMessage { msg ⇒
-          probe.ref ! s"$id $msg"
+          probe.ref ! Ack(msg, id)
           Behaviors.same
         }
       }))
@@ -52,11 +53,21 @@ class RoutersSpec extends ScalaTestWithActorTestKit("""
       val expectedStarted = (0 to 3).map { n ⇒ s"started $n" }.toSet
       probe.receiveMessages(4).toSet should ===(expectedStarted)
 
-      (0 to 8).foreach { n ⇒
-        pool ! s"message-$n"
-        val expectedRecipient = n % 4
-        probe.expectMessage(s"$expectedRecipient message-$n")
+      // send one message at a time and see we rotate over all children, noe that we don't necessarily
+      // know what order the logic is rotating over the children, so we just check we reach all of them
+      val sent = (0 to 8).map { n ⇒
+        val msg = s"message-$n"
+        pool ! msg
+        msg
       }
+
+      val acks = (0 to 8).map(_ ⇒ probe.expectMessageType[Ack])
+      val (recipients, messages) = acks.foldLeft[(Set[Int], Set[String])]((Set.empty, Set.empty)) {
+        case ((recipients, messages), ack) ⇒
+          (recipients + ack.recipient, messages + ack.msg)
+      }
+      recipients should ===(Set(0, 1, 2, 3))
+      messages should ===(sent.toSet)
     }
 
     "keep routing to the rest of the children if some children stops" in {

--- a/akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/RoutersSpec.scala
+++ b/akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/RoutersSpec.scala
@@ -53,7 +53,7 @@ class RoutersSpec extends ScalaTestWithActorTestKit("""
       val expectedStarted = (0 to 3).map { n ⇒ s"started $n" }.toSet
       probe.receiveMessages(4).toSet should ===(expectedStarted)
 
-      // send one message at a time and see we rotate over all children, noe that we don't necessarily
+      // send one message at a time and see we rotate over all children, note that we don't necessarily
       // know what order the logic is rotating over the children, so we just check we reach all of them
       val sent = (0 to 8).map { n ⇒
         val msg = s"message-$n"

